### ISSUE TITLE
Koji vs faster-routes

### ISSUE THREAD

**Issue by kamieniarz** [TurtIeSocks/Koji#116]
Hey. I just wanted to leave my observations here. I'm using faster-routes https://github.com/ccev/faster-routes for months so I wanted to make test if koji is better in generating route. Below are my results from a 7.5 km^2 area with 5039 spawnpoints

| -  | Koji | faster-routes |
| ------------- | ------------- | ----- |
| Calculation time  | ~3 sec  | ~13 min |
| Route length  | 550 | 505 |

Koji settings:
![image](https://user-images.githubusercontent.com/9063907/226186833-464b2d6d-a0e4-4bf0-8885-b6ff7731d995.png)

faster-routes settings:
![image](https://user-images.githubusercontent.com/9063907/226186845-8e20324d-d21c-40a4-8c28-740bc2471344.png)

I set same last seen values so number of spawnpoints for both tools was the same

I ran similar tests with even 10x bigger areas and results were the same - faster-routes generates ~10% shorter route with the same coverage. Below are screenshots from these routes - koji one looks to me like bootstrap with excluded parts without spawnpoints when faster-routes seems to be more optimized.

Koji:
![image](https://user-images.githubusercontent.com/9063907/226187135-3848292b-cd6c-4890-8ac7-76890bb46a3d.png)

faster-routes:
![image](https://user-images.githubusercontent.com/9063907/226187105-bbd8619a-37c6-4f29-9656-5ab9efcc6094.png)

So basically looks like I'm doing something wrong or maybe there's still more to be done ;) Anyway the speed is impressive tho

**Comment by kamieniarz** [comment]
I'm back with results after merging https://github.com/TurtIeSocks/Koji/pull/132 ;)

Same area, 5009 spawnpoints
Faster-routes: 505 routepoints (still)
Koji (brute force, cluster split level 10): 542 routepoints
Koji (fast): 664 routepoints
Koji (balanced): 552 routepoints - seems pretty much the same as in first post where it was 550

Honestly no idea what cluster split level does tho. I changed it to 1 and result was the same but with 15 it was 593

Here's a screenshot of Koji's route (the first one, brute force) which looks more like faster-route's one, so it's not that regular as it was before (or as it still is now when using fast or balanced cluster method). But this time I see a lot of spawnpoints out of routepoints...
![image](https://user-images.githubusercontent.com/9063907/234105303-c6acf1b5-1eb4-4ca6-b1f3-fad457acbe95.png)

So yeah, brute force looks better but it's still not *it*, at least when it comes to same settings as in first post -> min points = 1. Basically from my point of view it's not better cause maybe route is shorter but it missed some spots which is not good. But visually route looks better so I hope that's a good sign ;)

**Comment by Jebula999** [comment]
Just adding on to this.
Can confirm "Brute Force" set to 1 "Min Points" leaving out quite a few points.

![image](https://user-images.githubusercontent.com/11380895/234193270-b40066eb-caf5-423d-aff3-5f1d22cc9e11.png)

**Comment by TurtIeSocks** [comment]
@kamieniarz @Jebula999 can you test #200, please?

**Comment by TurtIeSocks** [comment]
Thanks for the tests, glad Koji is finally putting out some good numbers! 

The routing algorithms shouldn't have changed. What do you have `sort_by` (or `Sort By` in UI) set to? You'll need to use `TSP` to take advantage of OR-Tools and get proper routing.

**Comment by kamieniarz** [comment]
O have geo-hash (proly default?). I'll play with that and see how it goes, lots of options I see there ;) Thanks a lot! Since I'm the OP I'm ready to close this issue but I'll leave it open for some time cause maybe others may have some observations
